packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts

```diff
@@ -84,7 +84,7 @@ export type GraphChatListClient = Pick<MessageThreadProps, 'userId'> & {
   moreChatThreadsToLoad: boolean | undefined;
   internalSelectedChat: GraphChatThread | undefined;
   internalPrevSelectedChat: GraphChatThread | undefined;
-  initialSelectedChatSet: boolean;
+  fireOnSelected: boolean; // takes care of a case when we first init ChatList and sets the selected chat Id but onselected is not fired.
```

Comment:
Why would you not want to fire onSelected() when the chat is selected the first time?

Reply:
OnSelected is fired whenever a chat is selected, regardless of first time or not. The comment is trying to indicate the case why we need this flag to help us figure out if a ChatId is set initially in the ChatList. 

When ChatList is loaded with a ChatId for the first time, at the time this is set, there are no chat threads yet. The right time to set ChatId is when chat threads are all loaded. This also signals for us to fire onSelected. Hence, we are using this flag for that purpose.